src/pages/listing/video-listing.js

```diff
@@ -0,0 +1,74 @@
+import { useState, useEffect } from "react";
+import { Loader, SmallNav, VideoCard } from "../../components";
+import { getData, useData } from "../../helpers";
+export function VideoListingPage() {
+  const { loader, videos, categories, setCategories, setVideos, setLoader } =
+    useData();
+  const [selectedVideos, setSelectedVideos] = useState(videos);
+  const [selectedCategory, setSelectedCategory] = useState("All");
+  let showVideos = selectedVideos.length === 0 ? videos : selectedVideos;
+  useEffect(
+    () =>
+      (videos.length === 0 || categories.length === 0) &&
+      (async () => {
+        setLoader(true);
+        const apiData = await getData();
+        setLoader(false);
+        setCategories(apiData[1].data.categories);
+        setVideos(apiData[0].data.videos);
+      })(),
+    []
+  );
+  return (
+    <div className="flex flex-column" style={{ flex: 1 }}>
+      <SmallNav
+        setSelectedVideos={setSelectedVideos}
+        setSelectedCategory={setSelectedCategory}
+      />
+      <h1 className="text-white text-center" style={{ margin: "1rem" }}>
+        {" "}
```

Comment:
Remove this comment as sometime it can cause random bugs

Reply:
Yeah VS code adds it automatically. Should look for a way around.